compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1647,25 +1647,19 @@ impl<'context> Elaborator<'context> {
             self.push_err(ResolverError::AbiAttributeOutsideContract { span });
         }
 
-        if !let_stmt.comptime && matches!(let_stmt.pattern, Pattern::Mutable(..)) {
+        let explicit_comptime = let_stmt.explicit_comptime;
+        if !explicit_comptime && matches!(let_stmt.pattern, Pattern::Mutable(..)) {
             let span = let_stmt.pattern.span();
             self.push_err(ResolverError::MutableGlobal { span });
         }
 
-        let comptime = let_stmt.comptime;
-
-        let (let_statement, _typ) = if comptime {
-            self.elaborate_in_comptime_context(|this| this.elaborate_let(let_stmt, Some(global_id)))
-        } else {
-            self.elaborate_let(let_stmt, Some(global_id))
-        };
+        let (let_statement, _typ) = self
+            .elaborate_in_comptime_context(|this| this.elaborate_let(let_stmt, Some(global_id)));
```

Comment:
If I'm understanding this correctly, now all globals are elaborated at comptime, right?

Reply:
Yes, that's correct